Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneUtility.java

```diff
@@ -536,40 +538,49 @@ public void indexConcepts(String userName)
 
     private void buildQuery(BooleanClause.Occur occur, PerFieldAnalyzerWrapper perFieldAnalyzerWrapper,
             QueryBuilder qBuild, BooleanQuery.Builder builder, LuceneField luceneFieldAnnotation, String searchString) {
-        if (luceneFieldAnnotation.isTokenized()) {
-            QueryBuilder shortTermsQueryBuilder = new QueryBuilder(perFieldAnalyzerWrapper);
-            BooleanQuery.Builder shortTermBooleanQueryBuilder = new BooleanQuery.Builder();
-            String[] searchStrings = searchString.split(" ");
-            for (String searchValue : searchStrings) {
-                Query q1 = shortTermsQueryBuilder.createPhraseQuery(luceneFieldAnnotation.lucenefieldName(),
-                        searchValue);
-                if (q1 != null) {
-                    // Q1 can be null when the search string is very
-                    // short such as "be", but the search can be
-                    // performed using the BooleanQuery q2.
-                    if (luceneFieldAnnotation.isShortPhraseSearchable()) {
-                        shortTermBooleanQueryBuilder.add(q1, BooleanClause.Occur.SHOULD);
-                    } else {
-                        shortTermBooleanQueryBuilder.add(q1, occur);
-                    }
+        if (luceneFieldAnnotation.isTokenized() && luceneFieldAnnotation.isShortPhraseSearchable()) {
+            BooleanQuery.Builder rootQueryBuilder = new BooleanQuery.Builder();
+            BooleanQuery.Builder tokenizedQueryBuilder = new BooleanQuery.Builder();
+            buildTokenizedOrWildCardQuery(luceneFieldAnnotation, searchString, tokenizedQueryBuilder);
+            rootQueryBuilder.add(tokenizedQueryBuilder.build(), Occur.SHOULD);
+
+            // Short word searching
+            BooleanQuery.Builder shortWordSearchQueryBuilder = new BooleanQuery.Builder();
+            shortWordSearchQueryBuilder
+                    .add(new PhraseQuery(luceneFieldAnnotation.lucenefieldName() + LuceneFieldNames.UNTOKENIZED_SUFFIX,
+                            searchString), Occur.SHOULD);
+            rootQueryBuilder.add(shortWordSearchQueryBuilder.build(), Occur.SHOULD);
+            // Adding both tokenized and untokenized queries to main query
+            // builder.
+            builder.add(rootQueryBuilder.build(), occur);
+        }
 
-                }
-            }
+        else if (luceneFieldAnnotation.isTokenized()) {
+            // Tokenized and not short word searchable
+            BooleanQuery.Builder tokenizedQueryBuilder = new BooleanQuery.Builder();
+            buildTokenizedOrWildCardQuery(luceneFieldAnnotation, searchString, tokenizedQueryBuilder);
+            builder.add(tokenizedQueryBuilder.build(), occur);
+        }
```

Comment:
this block can be merged with the previous if block. just the last line is different.